app/services/procedure_export_v2_service.rb

```diff
@@ -0,0 +1,84 @@
+class ProcedureExportV2Service
+  attr_reader :dossiers
+
+  def initialize(procedure, ids: nil, since: nil, limit: nil)
+    @procedure = procedure
+    @dossiers = procedure.dossiers.downloadable_sorted
+    if ids
+      @dossiers = @dossiers.where(id: ids)
+    end
+    if since
+      @dossiers = @dossiers.since(since)
+    end
+    if limit
+      @dossiers = @dossiers.limit(limit)
+    end
+    @tables = [:dossiers, :etablissements, :avis] + repetitions.map do |champ|
+      [
+        champ.libelle,
+        champ.rows.each_with_index.map do |champs, index|
+          Champs::RepetitionChamp::Row.new(index: index + 1, dossier_id: champ.dossier_id.to_s, champs: champs)
+        end
+      ]
+    end
+  end
+
+  def to_csv(table = :dossiers)
+    SpreadsheetArchitect.to_csv(options_for(table))
+  end
+
+  def to_xlsx
+    # We recursively build multi page spreadsheet
+    @tables.reduce(nil) do |package, table|
+      SpreadsheetArchitect.to_axlsx_package(options_for(table), package)
+    end.to_stream.read
+  end
+
+  def to_ods
+    # We recursively build multi page spreadsheet
+    @tables.reduce(nil) do |spreadsheet, table|
+      SpreadsheetArchitect.to_rodf_spreadsheet(options_for(table), spreadsheet)
+    end.bytes
+  end
+
+  private
+
+  def etablissements
+    @etablissements ||= dossiers.flat_map do |dossier|
+      [dossier.champs, dossier.champs_private]
+        .flatten
+        .select { |champ| champ.is_a?(Champs::SiretChamp) }
+    end.map(&:etablissement).compact + dossiers.map(&:etablissement).compact
+  end
+
+  def avis
+    @avis ||= dossiers.flat_map(&:avis)
+  end
+
+  def repetitions
```

Comment:
```suggestion
  def champs_repetables
```

Pour que ça soit plus clair quand on écrit `@tables = [:dossiers, :etablissements, :avis] + champs_repetables` ?